### ISSUE TITLE
Start and Stop Screen Capture for ProPresenter 7

### ISSUE THREAD

**Issue by jhollowell41** [bitfocus/companion-module-renewedvision-propresenter#70]
Feature request for buttons to start and stop screen capture recording for ProPresenter 7.

**Comment by josephdadams** [comment]
@greyshirtguy possible?

**Comment by greyshirtguy** [comment]
Not presently.  There is no support for those actions in either the remote app (which is the same protocol we use) or even with MIDI !

**Comment by greyshirtguy** [comment]
I've been thinking about building a "helper" app that allows Companion to do more with ProPresenter.

Something that you run on the computer along side ProPresenter, controlled by the Companion module, and it can be told to perform actions like clicking on menus and buttons in the actual UI of Pro7...

I'm in two minds, because it would use methods that are "hackish" and "brittle" making it easily broken when Pro7 is updated....It coudl very quickly become a support burden every time a new update comes out -I would have to scramble to get all hacks working again....

**Comment by jhollowell41** [comment]
If propresenter/renewed vision  created a keyboard shortcut, would it then be possible?  If so, I can place a feature request with them. 

Jody

Please excuse the brevity and/or typos, as this message was sent from my iPhone.

**Comment by greyshirtguy** [comment]
Yes, however, a much better feature to request would be to request to control streaming and recroding from the remote app.
Whatever the remote app can do, we can reverse-engineer and add to Companion (and unlike hotkeys, the remote app works even when Pro7 does _not_ have keyboard focus)

**Comment by josephdadams** [comment]
An actual published API would be nice too, so that when they arbitrarily make a change for their remote app, it doesn't break Companion support and confuse a bunch of users who have come to rely on that reverse engineered protocol that makes Companion work with it.

**Comment by greyshirtguy** [comment]
Hang in there everyone....it may take a while but I am confident good news is on it's way....

**Comment by mrhall89** [comment]
Any update on this? Capture is supported within the API and can be utilized easily via Companion's included HTTP connection. Would be more convenient to see it in the ProPresenter connection though.

**Comment by greyshirtguy** [comment]
The API totally support this a a small # of other features not possible with old reversed eng API..
The current module supports Pro6 and early Pro7 using this old API..
I have chosen to make a new module that is new API only (and only works with Pro 7.9 or higher).
I hope to have this new module added to the 3 beta in the next coupl eof months!
Until then, the generic HTTP module is very easy to use and make any new API call....
